File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/mappers/elementcache/ApiElementCacheShareMapper.java
package com.gigigo.orchextra.core.data.mappers.elementcache;

import com.gigigo.orchextra.core.data.dto.elementcache.ApiElementCacheShare;
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheShare;
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;

/**
 * Created by rui.alonso on 5/12/16.
 */
public class ApiElementCacheShareMapper implements
    ExternalClassToModelMapper<ApiElementCacheShare, ElementCacheShare> {

  @Override public ElementCacheShare externalClassToModel(ApiElementCacheShare data) {
    ElementCacheShare model = new ElementCacheShare();

    model.setUrl(data.getUrl());
    model.setText(data.getText());

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elementcache/ApiElementCacheShareMapper.kt
package com.gigigo.orchextra.core.data.api.mappers.elementcache

import android.util.Log
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elementcache.ApiElementCacheShare
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheShare

class ApiElementCacheShareMapper : ExternalClassToModelMapper<ApiElementCacheShare, ElementCacheShare> {

  override fun externalClassToModel(data: ApiElementCacheShare): ElementCacheShare {
    val time = System.currentTimeMillis()

    val model = ElementCacheShare()

    with(model) {
      url = data.url
      text = data.text
    }

    val currentTime = System.currentTimeMillis() - time
    Log.v("TT - ApiElemCacheShare", ("" + currentTime/1000))

    return model
  }
}
